crates/rspack_core/src/compiler/compilation.rs

```diff
@@ -930,34 +935,36 @@ impl Compilation {
       }
       jobs.extend(map.into_values());
     }
-    let results = jobs
-      .into_par_iter()
-      .map(|job| {
-        let module = job.module;
-        (
-          module,
-          self
-            .old_cache
-            .code_generate_occasion
-            .use_cache(job, |module, runtime| {
-              let module = module_graph
-                .module_by_identifier(&module)
-                .expect("should have module");
-              module
-                .code_generation(self, Some(runtime), None)
-                .map(|mut codegen_res| {
-                  codegen_res.set_hash(
-                    &self.options.output.hash_function,
-                    &self.options.output.hash_digest,
-                    &self.options.output.hash_salt,
-                  );
-                  codegen_res
-                })
-            }),
-        )
-      })
-      .collect::<Vec<_>>();
-    for (module, (codegen_res, runtimes, from_cache)) in results {
+
+    let results = join_all(jobs.into_iter().map(|job| async {
```

Comment:
this change parallel to concurrency which seems will cause performance regression consider code_gen is cpu heavy task